slang-gfx.h

```diff
@@ -668,14 +668,14 @@ class ITextureResource: public IResource
 
 enum class ComparisonFunc : uint8_t
 {
-    Never           = 0,
-    Less            = 0x01,
-    Equal           = 0x02,
-    LessEqual       = 0x03,
-    Greater         = 0x04,
-    NotEqual        = 0x05,
-    GreaterEqual    = 0x06,
-    Always          = 0x07,
+    Never           = 0x0,
+    Less            = 0x1,
+    Equal           = 0x2,
+    LessEqual       = 0x3,
+    Greater         = 0x4,
+    NotEqual        = 0x5,
+    GreaterEqual    = 0x6,
+    Always          = 0x7,
```

Comment:
Unrelated to this PR: at some point this warrants a block comment about how the cases here and their numeric values are actually cleverly arranged, using a scheme that dates at least back to the origins of OpenGL (and thus probably earlier).

The cases for comparisons in OpenGL have values such that `LessEqual = Less | Equal`, etc. Bitwise operations on the comparison functions correspond to logical operations on the conditions. One can take an unknown condition `c` and form the condition that matches in the same cases except *not* on equality with `c & ~Equal` or `c & NotEqual`.

This same style of enumeration can be extended to include an additional bit for `Unordered` to handle floating-point types, which don't have a total order. I don't think any graphics APIs currently support unordered comparisons in fixed-function units, but it is a relevant thing when setting up things like comparison ops in a compiler IR.

Documenting this kind of thing isn't the most important thing in the world, but it can help pass on knowledge that might otherwise get lost in successive generations of systems. For example, the D3D11/12 equivalents of this enumeration have the same cases, in the same order, but do *not* have the same properties described above because the `D3D11_NEVER`/`D3D12_NEVER` cases start at a value of `1` instead of `0` (presumably to leave `0` as an out-of-band value).